examples/step-9/doc/intro.dox

```diff
@@ -53,27 +53,70 @@ be negative if the transport direction $\beta$ points inward, i.e. at
+for the derivatives in the perpendicular direction, and consequently
+if $u$ is discontinuous at a point on the inflow boundary, then this
+discontinuity will simply be transported along the streamline of the
+wind field that starts at this boundary point.
+
+These discontinuities lead to numerical instabilities that
 make a stable solution by a straight-forward discretization
 impossible. We will thus use the streamline diffusion stabilized
 formulation, in which we test the equation with test functions $v +
 \delta \beta\cdot\nabla v$ instead of $v$, where $\delta$ is a
 parameter that is chosen in the range of the (local) mesh width $h$;
 good results are usually obtained by setting $\delta=0.1h$. Note that
 the modification in the test function vanishes as the mesh size tends
-to zero. We will not discuss reasons, pros, and cons of the streamline
+to zero. The modification in the test function can be thought as
+adding a small amount of diffusion along the direction $\beta$ of the
+wind field, i.e., along "stream lines".
+We will not discuss reasons, pros, and cons of the streamline
 diffusion method, but rather use it "as is", and refer the
 interested reader to the sufficiently available literature; every
 recent good book on finite elements should have a discussion of that
 topic.
 
-Using the test functions as defined above, the weak formulation of
-our stabilized problem reads: find a discrete function $u_h$ such that
+Using the test functions as defined above, an initial weak form of the
+problem would ask for finding a function $u_h$ so that for all test
+functions $v_h$ we have
+@f[
+  (\beta \cdot \nabla u_h, v_h + \delta \beta\cdot\nabla v_h)_\Omega
+  =
+  (f, v_h + \delta \beta\cdot\nabla v_h)_\Omega.
+@f]
+However, we would like to include inflow boundary conditions $u=g$
+weakly into this problem, and this can be done by requiring that in
+addition to the equation above we also have
+@f[
+  (u_h, w_h)_{\partial\Omega_-}
+  =
+  (g, w_h)_{\partial\Omega_-}
+@f]
+for all test functions $w_h$ that live on the boundary and that are
+from a suitable test space. It turns out that a suitable space of test
+functions happens to be $\beta\cdot {\mathbf n}$ times the traces of
+the functions $v_h$ in the test space we already use for the
+differential equation in the domain. Thus, we require that for all
+test functions $v_h$ we have
+@f[
+  (u_h, \beta\cdot {\mathbf n} v_h)_{\partial\Omega_-}
+  =
+  (g, \beta\cdot {\mathbf n} v_h)_{\partial\Omega_-}.
+@f]
+Without attempting a justification (see again the literature on the finite
+element method in general, and the streamline diffusion method in
+particular), we can combine the equations for the differential
+equation and the boundary values in the following
+weak formulation of
+our stabilized problem: find a discrete function $u_h$ such that
 for all discrete test functions $v_h$ there holds
 @f[
   (\beta \cdot \nabla u_h, v_h + \delta \beta\cdot\nabla v_h)_\Omega
```

Comment:
should this be 
```
(u_h, \beta\cdot {\mathbf n} v_h)_{\partial\Omega_-}
```
instead of
```
(\beta\cdot {\mathbf n} u_h, v_h)_{\partial\Omega_-}
```
(same for RHS)?

Reply:
Note that this is not a line you changed...

Reply:
But that's the same thing, right? Or what am I missing?

Reply:
I am switching u_h and v_h

Reply:
Hm, I'm confused. Your suggestion is attached to a line that doesn't even have the term you mention. Which line are you talking about?

In the end, your suggestion is a no-op -- you're just moving the weight factor `\beta \cdot \mathbf n` from one side of the scalar product to the other.

Reply:
It is the line below the one I attached to. Go to http://www.dealii.org/developer/doxygen/deal.II/step_9.html and search for "there holds". Notice the equation directly below incorporates the equation (you added) but has u_h and v_h switched. That doesn't make sense to me.

> In the end, your suggestion is a no-op

That is true but it still confused me. There is no reason to move it over.

Reply:
I see. I don't feel strongly one way or the other -- so see #5287.